ngraph/core/include/ngraph/op/util/gather_base.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace util
+        {
+            /// \brief GatherBase basic class for Gather v1 and v7
+            class NGRAPH_API GatherBase : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                GatherBase() = default;
+
+                /// \param data The tensor from which slices are gathered
+                /// \param indices Tensor with indexes to gather
+                /// \param axis The tensor is a dimension index to gather data from
+                /// \param batch_dims The number of batch dimension in data and indices tensors
+                GatherBase(const Output<Node>& data,
+                           const Output<Node>& indices,
+                           const Output<Node>& axis,
+                           const int64_t batch_dims = 0);
+
+                void validate_and_infer_types() override;
+                int64_t get_batch_dims() const;
```

Comment:
Why we have `get_batch_dims` method in the base class? It is not applicable for Gather1. 

@GlebKazantaev , are you ok with that?